### DISCUSSION TITLE
How can I get the full/absolute URL of NetBox instance within a NetBox script?

### DISCUSSION THREAD

**Discussion by drygdryg** [netbox-community/netbox#7054] [Q&A] [answered]
Can I get URL of my NetBox instance from [a NetBox custom script](https://netbox.readthedocs.io/en/stable/additional-features/custom-scripts/)?

**Comment by candlerb** [comment, marked as answer]
In general, your Netbox instance may be accessed via many URLs which all resolve to the same host.

Hence you probably want to look at the Request object to see how the user is currently accessing it:
https://netbox.readthedocs.io/en/stable/additional-features/custom-scripts/#accessing-request-data

Look at how the SVG code creates an absolute base URL:
https://github.com/netbox-community/netbox/blob/v2.11.12/netbox/dcim/api/views.py#L216

Or you may be able to use the django 'reverse' method:
https://github.com/netbox-community/netbox/blob/v2.11.12/netbox/dcim/models/devices.py#L643-L644
